log4j-core/src/main/java/org/apache/logging/log4j/core/config/PropertiesPlugin.java

```diff
@@ -45,15 +46,14 @@ private PropertiesPlugin() {
     @PluginFactory
     public static StrLookup configureSubstitutor(@PluginElement("Properties") final Property[] properties,
                                                  @PluginConfiguration final Configuration config) {
-        if (properties == null) {
-            return new Interpolator(config.getProperties());
-        }
+        final Property[] props = properties == null ? Property.EMPTY_ARRAY : properties;
         final Map<String, String> map = new HashMap<>(config.getProperties());
```

Comment:
TODO + test: the configuration context-properties (`config.getProperties()`) are distinct from the configuration properties (`Property[] properties`), and should not allow recursive evaluation.
The PropertiesLookup should take both maps, and return values which don't support recursive evaluation for config context props.